jdaviz/app.py

```diff
@@ -1304,14 +1309,33 @@ def is_there_overlap_spectral_subset(self, subset_name):
         Returns True if the spectral subset with subset_name has overlapping
         subregions.
         """
-        spectral_region = self.get_subsets(subset_name, spectral_only=True)
-        if not spectral_region or len(spectral_region) < 2:
+        spectral_region = self.get_subsets(subset_name, simplify_spectral=False, spectral_only=True)
+        n_reg = len(spectral_region)
+        if not spectral_region or n_reg < 2:
             return False
-        for index in range(0, len(spectral_region) - 1):
-            if spectral_region[index].upper.value >= spectral_region[index + 1].lower.value:
+        for index in range(n_reg - 1):
+            if spectral_region[index]['region'].upper.value >= spectral_region[index + 1]['region'].lower.value:  # noqa: E501
                 return True
         return False
 
+    @staticmethod
+    def _merge_overlapping_spectral_regions_worker(spectral_region):
```

Comment:
Same logic as before but I find operating on SpectralRegion directly less complicated than forcing tuple of floats. Also I need SpectralRegion directly to reuse it somewhere else.